### PR TITLE
Change metadata script syntax

### PR DESCRIPTION
**Pull request by Kathrin-Huber**
This changes unintuitive syntax to new:

Example:
`action:addData key:titleDocMain value:myTitle`

possible attributes:
* key (name of new key), 
* value (new value), 
* source (when value is copied from another metadatakey), 
* variable (e.g. processid, all from VariableReplacer), 
* type (if new metadata should be added to a specific type, e.g. PeriodicalIssue)

### CODE REVIEW

File: Kitodo/src/main/java/org/kitodo/production/services/command/MetadataScript.java [hunk 1]
```diff
@@ -12,60 +12,62 @@
...
-        }
-        if (metadataCommand.contains("=")) {
-            String[] metadataParts = metadataCommand.split("=");
-            goal = metadataParts[0];
-            String rootOrValue = metadataParts[1];
-            if (rootOrValue.startsWith("@") || rootOrValue.startsWith("$")) {
-                root = rootOrValue;
-            } else {
-                values.add(rootOrValue);
+        this.parameters = new HashMap<>();
```
Comment (matthias-ronge) [inline Kitodo/src/main/java/org/kitodo/production/services/command/MetadataScript.java:36]: `parameters` isn’t used outside of the function. Please replace the class field by a local variable.

File: Kitodo/src/main/java/org/kitodo/production/services/command/MetadataScript.java [hunk 2]
```diff
@@ -76,19 +78,19 @@ public String getRoot() {
...
      */
     public String getTypeTarget() {
         return typeTarget;
     }
+
+    public String getVariable() {
+        return variable;
+    }
+
+    public void setVariable(String variable) {
```
Comment (matthias-ronge) [inline Kitodo/src/main/java/org/kitodo/production/services/command/MetadataScript.java:93]: The setter is not used. You can delete it, to make `variable` final.

File: Kitodo/src/main/java/org/kitodo/production/services/command/MetadataScript.java [hunk 3]
```diff
@@ -12,60 +12,62 @@
...
 public class MetadataScript {
 
-    private String goal;
-    private String root;
+    private Map<String, String> parameters;
+    private String metadataKey;
+    private String valueSource;
     private String typeTarget;
+    private String variable;
     private List<String> values = new ArrayList<>();
```
Comment (matthias-ronge) [inline Kitodo/src/main/java/org/kitodo/production/services/command/MetadataScript.java:29]: `values` is always zero or at most one element. Please change to `Optional<String> value;`

File: Kitodo/src/main/java/org/kitodo/production/services/command/MetadataScript.java [hunk 4]
```diff
@@ -12,60 +12,62 @@
...
 import java.util.Objects;
 
+import org.apache.commons.lang.text.StrTokenizer;
+
 public class MetadataScript {
 
-    private String goal;
-    private String root;
+    private Map<String, String> parameters;
+    private String metadataKey;
```
Comment (matthias-ronge) [inline Kitodo/src/main/java/org/kitodo/production/services/command/MetadataScript.java:25]: It looks like you can make all of these variables `final`.

File: Kitodo/src/main/java/org/kitodo/production/services/command/MetadataScript.java [hunk 5]
```diff
@@ -12,60 +12,62 @@
...
             }
         }
-        else {
-            goal = command;
-        }
-        if (typeCommand.contains("=")) {
-            typeTarget = typeCommand.split("=")[1];
+
+        metadataKey = parameters.get("key");
+        if (Objects.nonNull(parameters.get("value"))) {
```
Comment (matthias-ronge) [inline Kitodo/src/main/java/org/kitodo/production/services/command/MetadataScript.java:48]: With `Optional<String> value`, you can simplify this line to:

    value = Optional.ofNullable(parameters.get("value"));

File: Kitodo/src/main/java/org/kitodo/production/services/command/AddDataScript.java [hunk 6]
```diff
@@ -36,7 +36,7 @@ public void executeScript(LegacyMetsModsDigitalDocumentHelper metadataFile, Proc
 
         for (String value : metadataScript.getValues()) {
```
Comment (matthias-ronge) [inline Kitodo/src/main/java/org/kitodo/production/services/command/AddDataScript.java:37]: `value` can occur at most once (see comment on `MetadataScript` class, replacement with `Optional<String>`). You can remove the loop.

File: Kitodo/src/main/java/org/kitodo/production/services/command/AddDataScript.java [hunk 7]
```diff
@@ -36,7 +36,7 @@ public void executeScript(LegacyMetsModsDigitalDocumentHelper metadataFile, Proc
...
             MetadataEntry metadataEntry = new MetadataEntry();
-            metadataEntry.setKey(metadataScript.getGoal());
+            metadataEntry.setKey(metadataScript.getMetadataKey());
             metadataEntry.setValue(value);
             metadataEntry.setDomain(MdSec.DMD_SEC);
```
Comment (matthias-ronge) [inline Kitodo/src/main/java/org/kitodo/production/services/command/AddDataScript.java:41]: I would suggest making the domain configurable as a parameter, or better yet, load the ruleset and look it up.

File: Kitodo/src/main/java/org/kitodo/production/services/command/MetadataScript.java [hunk 8]
```diff
@@ -12,60 +12,62 @@
...
-            String[] metadataParts = metadataCommand.split("=");
-            goal = metadataParts[0];
-            String rootOrValue = metadataParts[1];
-            if (rootOrValue.startsWith("@") || rootOrValue.startsWith("$")) {
-                root = rootOrValue;
-            } else {
-                values.add(rootOrValue);
+        this.parameters = new HashMap<>();
+        StrTokenizer tokenizer = new StrTokenizer(command, ' ', '\"');
+        while (tokenizer.hasNext()) {
```
Comment (matthias-ronge) [inline Kitodo/src/main/java/org/kitodo/production/services/command/MetadataScript.java:38]: The script is already tokenized in the `KitodoScriptService` class. The result is available there in the class field `parameters` of the class `KitodoScriptService`. Please do not tokenize the script here again, instead, pass the `parameters` field through here.

**Review by matthias-ronge — CHANGES_REQUESTED** [review]

File: Kitodo/src/main/java/org/kitodo/production/services/command/MetadataScript.java (hunk 8 above)
Comment (Kathrin-Huber) [reply Kitodo/src/main/java/org/kitodo/production/services/command/MetadataScript.java:38]: It's not parsed correctly in the KitodoScriptService. I cannot use it, when multiple scripts are entered with ';'

**Pull request merged**